app/assets/javascripts/discourse/lib/highlight-html.js

```diff
@@ -0,0 +1,83 @@
+function highlight(node, pattern, nodeName, className) {
+  if (
+    ![Node.ELEMENT_NODE, Node.TEXT_NODE].includes(node.nodeType) ||
+    ["SCRIPT", "STYLE"].includes(node.tagName) ||
+    (node.tagName === nodeName && node.className === className)
+  ) {
+    return 0;
+  }
+
+  if (node.nodeType === Node.ELEMENT_NODE && node.childNodes) {
+    for (let i = 0; i < node.childNodes.length; i++) {
+      i += highlight(node.childNodes[i], pattern, nodeName, className);
+    }
+    return 0;
+  }
+
+  if (node.nodeType === Node.TEXT_NODE) {
+    const match = node.data.match(pattern);
+
+    if (!match) {
+      return 0;
+    }
+
+    const element = document.createElement(nodeName);
+    element.className = className;
+    element.innerText = match[0];
+    const matchNode = node.splitText(match.index);
+    matchNode.splitText(match[0].length);
+    matchNode.parentNode.replaceChild(element, matchNode);
+    return 1;
+  }
+
+  return 0;
+}
+
+export default function(node, words, opts = {}) {
+  let settings = {
+    nodeName: "span",
+    className: "highlighted",
+    matchCase: false
+  };
+
+  Object.assign(settings, opts);
+  words = typeof words === "string" ? [words] : words;
```

Comment:
<img width="197" alt="Screenshot 2020-04-03 at 15 21 16" src="https://user-images.githubusercontent.com/339945/78365034-c6eedb80-75be-11ea-88c2-62b5b9e4b27f.png">

We have `makeArray` for this pattern